service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyService.java

```diff
@@ -54,16 +62,24 @@ private void checkForTrailingSlashes(ServiceContext serviceContext) {
         Iterator<URI> connectIterator = connectURIs.iterator();
 
         while (acceptIterator.hasNext() && connectIterator.hasNext()) {
-            String acceptPath = acceptIterator.next().getPath();
-            String connectPath = connectIterator.next().getPath();
-            boolean acceptPathEndsInSlash = acceptPath.endsWith("/");
-            boolean connectPathEndsInSlash = connectPath.endsWith("/");
+            URI acceptURI = acceptIterator.next();
+            URI connectURI = connectIterator.next();
+            String acceptPath = acceptURI.getPath();
+            String connectPath = connectURI.getPath();
+            boolean acceptPathIsSlash = "/".equals(acceptPath);
+            boolean connectPathIsSlash = "/".equals(connectPath);
+            boolean acceptPathNoTrailingSlash = !acceptPath.endsWith("/") && acceptPath.length() > 1;
+            boolean connectPathNoTrailingSlash = !connectPath.endsWith("/") && connectPath.length() > 1;
 
-            if (acceptPathEndsInSlash != connectPathEndsInSlash) {
+            if (acceptPathIsSlash && connectPathNoTrailingSlash) {
+                throw new IllegalArgumentException(String.format(ACCEPT_PATH_IS_SLASH_ERROR_MESSAGE,
+                        serviceContext.getServiceName(), acceptURI, connectURI));
+            } else if (acceptPathNoTrailingSlash && connectPathIsSlash) {
+                throw new IllegalArgumentException(String.format(CONNECT_PATH_IS_SLASH_ERROR_MESSAGE,
+                        serviceContext.getServiceName(), acceptURI, connectURI));
+            } else if (acceptPathNoTrailingSlash != connectPathNoTrailingSlash) {
                 throw new IllegalArgumentException(
-                        "Please fix the gateway configuration file for service '" + serviceContext.getServiceName()
-                                + "'. Accept and connect must both either have trailing slashes,"
-                                + " or both must not end in slashes.");
+                        String.format(ACCEPT_CONNECT_ERROR_MESSAGE, serviceContext.getServiceName()));
```

Comment:
I would go with

```
    private void checkForTrailingSlashes(ServiceContext serviceContext) {
        Collection<URI> acceptURIs = serviceContext.getAccepts();
        Collection<URI> connectURIs = serviceContext.getConnects();

        assert acceptURIs.size() == 1;
        assert connectURIs.size() == 1;

        URI acceptURI = acceptURIs.iterator().next();
        URI connectURI = connectURIs.iterator().next();
        String acceptPath = acceptURI.getPath();
        String connectPath = connectURI.getPath();

        boolean acceptPathIsSlash = acceptPath.endsWith("/");
        boolean connectPathIsSlash = connectPath.endsWith("/");
        if (acceptPathIsSlash ^ connectPathIsSlash) {
            if (acceptPathIsSlash) {
                URI connectURIWithSlash = connectURI.resolve(connectPath + "/");
                throw new IllegalArgumentException("Configure connect URI as " + connectURIWithSlash + " since accept URI " + acceptURI + " ends with a trailing /");
            } else {
                String noTrailingSlash = connectPath.substring(0, connectPath.length()-1);
                URI connectURIWithNoSlash = connectURI.resolve(noTrailingSlash);
                throw new IllegalArgumentException("Configure connect URI as " + connectURIWithNoSlash + " since accept URI " + acceptURI + " doesn't end with a trailing /");
            }
        }
    }
```

If you want you can add additional ". Both accept and connect URIs either end with trailing / or not." to IllegalArgumentException.